### ISSUE TITLE
Memgraph can't start with volume in bind mode

### ISSUE THREAD

**Issue by vector-mj** [memgraph/memgraph#1734]
**Memgraph version**: v2.14.1

**Log**:
```sh
[2024-02-18 18:14:30.083] [warning] storage_parallel_index_recovery flag is deprecated. Check storage_mode_parallel_schema_recovery for more details.
[2024-02-18 18:14:30.083] [warning] storage_parallel_index_recovery flag is deprecated. Check storage_mode_parallel_schema_recovery for more details.
You are running Memgraph v2.14.1
To get started with Memgraph, visit https://memgr.ph/start
```
**Environment**
```sh
--log-level=TRACE
```

**My docker-compose file**
```yaml
version: '3.7'
services:
  memgraph:
    image: memgraph/memgraph
    ports:
      - 7687
      - 7444
      - 3000:3000
    command: ["--log-level=TRACE"]
    volumes:
      - ./data:/var/lib/memgraph
    deploy:
      resources:
        limits:
          cpus: '1'
          memory: 1G
```

**Description**
Memgraph can't start and doesn't have any output log even in the TRACE log level, but it works correctly if I change the bind type volume to docker volume.

**Comment by katarinasupe** [comment]
Hi @vector-mj, thank you for opening the issue. We know this is not working, probably due to file ownership issues. Is this a blocker for you currently or is the Docker volume workaround good enough for you for now?

**Comment by katarinasupe** [comment]
So far, we recommended changing the default location of a docker data directory with the `--data-root` [property](https://docs.docker.com/engine/reference/commandline/dockerd/ ). Here is more on that topic: 
https://stackoverflow.com/questions/36014554/how-to-change-the-default-location-for-docker-create-volume-command

Let me know if that helps somehow.

We also had suggestions from our Discord community for workarounds ([you can join there](https://www.discord.gg/memgraph)). For example, this is what a community member suggested for the Memgraph Platform (which can probably be applied to Memgraph image as well):
```
version: "3.9"
services:
  memgraph:
    image: memgraph/memgraph-platform:latest
    ports:
      - "7687:7687"
      - "3000:3000"
      - "7444:7444"
    volumes:
      - mg_lib:/var/lib
      - mg_log:/var/log
      - mg_etc:/etc/memgraph
    environment:
      - MEMGRAPH="--storage-properties-on-edges=true --log-level=TRACE"
    entrypoint: ["/usr/bin/supervisord"]
volumes:
  mg_lib:
    driver: local
    driver_opts:
      type: 'none'
      o: 'bind'
      device: 'path/on/host'
  mg_log:
    driver: local
    driver_opts:
      type: 'none'
      o: 'bind'
      device: 'path/on/host'
  mg_etc:
    driver: local
    driver_opts:
      type: 'none'
      o: 'bind'
      device: 'path/on/host'
```

Tbh, I haven't tried the above, and it should be tested, but I thought it's best I provide you with all the information I have. Here is the [thread on Discord](https://discord.com/channels/842007348272169002/1076084990191874069/1166696944358395995).

**Comment by vector-mj** [comment]
Thank you @katarinasupe, your Docker compose manifest worked for me.

```yaml
version: '3.9'
services:
  memgraph:
    image: docker.arvancloud.ir/memgraph/memgraph-platform
    container_name: memgraph
    ports:
      - 7687:7687
      - 7444:7444
      - 3000:3000
    environment:
      MEMGRAPH: "--memory-limit=1000 --log-level=TRACE --storage-properties-on-edges=true --storage-snapshot-interval-sec=2 --storage-snapshot-retention-count=3 --storage-mode=ON_DISK_TRANSACTIONAL --storage-snapshot-on-exit=true"
    volumes:
      - mg_lib:/var/lib
      - mg_log:/var/log
      - mg_etc:/etc/memgraph
    deploy:
      resources:
        limits:
          cpus: '1'
          memory: 1G
volumes:
  mg_lib:
    driver: local
    driver_opts:
      type: 'none'
      o: 'bind'
      device: './memgraph/data'
  mg_log:
    driver: local
    driver_opts:
      type: 'none'
      o: 'bind'
      device: './memgraph/log'
  mg_etc:
    driver: local
    driver_opts:
      type: 'none'
      o: 'bind'
      device: './memgraph/config'
```

**Comment by vector-mj** [comment]
> That's great to hear @vector-mj 😄 Btw., if you don't mind me asking, what are you working on with Memgraph?

Hi again @katarinasupe
I'm sorry to reply too late. I'm working on my personal social media project and after some research, I decided to use Memgraph instead of Neo4j for faster queries.

**Comment by katarinasupe** [comment]
Glad to hear you decided to use Memgraph 😄 If you have more questions or you'd like to share more about the project with the community, [join our Discord server](https://www.discord.gg/memgraph).

**Comment by kenho811** [comment]
I also got this issue where once I attached the volume, I cannot connect to memgraph server via memgraph-lab.

Using the docker-compose file provided above, I re-ran it several times and now it's working.

I further observe that it requires that the ownership of folder be like this

<img width="721" height="175" alt="Image" src="https://github.com/user-attachments/assets/10e058f4-df56-4515-94bc-84c9962a9a09" />

```
memgraph@memgraph-prod:~$ ls -lsrt
total 24
4 drwxr-xr-x 10 memgraph   memgraph 4096 Aug 23 13:56 home_data_centre
4 -rw-r--r--  1 memgraph   memgraph  939 Aug 23 17:04 docker-compose.yaml
4 drwxr-xr-x  4 root       root     4096 Aug 23 17:04 config
4 drwxr-x---  2 messagebus adm      4096 Aug 23 17:04 log
4 drwxr-x--- 14 messagebus kvm      4096 Aug 23 17:04 data
```

When I attempt to change the config/log/data back to the user `memgraph:memgraph` group in my VM, I got connection issue in memgraph-lab again.

So it looks like the folder ownership by messagebus with group adm/kvm is necessary.
